src/Compilers/CSharp/Test/Semantic/Semantics/QueryTests.cs

```diff
@@ -4134,15 +4134,15 @@ static void Main()
                   ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 3) (Syntax: '3')
 ";
             var expectedDiagnostics = new DiagnosticDescription[] {
-                // CS0119: 'ConsoleApp' is a namespace, which is not valid in the given context
+                // file.cs(13,35): error CS0119: 'ParentNamespace.ConsoleApp' is a namespace, which is not valid in the given context
                 //                 var x = from c in ConsoleApp select 3;
-                Diagnostic(ErrorCode.ERR_BadSKunknown, "ConsoleApp").WithArguments("ConsoleApp", "namespace").WithLocation(13, 35),
-                // CS0119: 'ParentNamespace.ConsoleApp' is a namespace, which is not valid in the given context
+                Diagnostic(ErrorCode.ERR_BadSKunknown, "ConsoleApp").WithArguments("ParentNamespace.ConsoleApp", "namespace").WithLocation(13, 35),
+                // file.cs(14,35): error CS0119: 'ParentNamespace.ConsoleApp' is a namespace, which is not valid in the given context
                 //                 var y = from c in ParentNamespace.ConsoleApp select 3;
                 Diagnostic(ErrorCode.ERR_BadSKunknown, "ParentNamespace.ConsoleApp").WithArguments("ParentNamespace.ConsoleApp", "namespace").WithLocation(14, 35),
-                // CS0119: 'NSAlias' is a namespace, which is not valid in the given context
+                // file.cs(15,45): error CS0119: 'ParentNamespace.ConsoleApp' is a namespace, which is not valid in the given context
                 //                 var z = /*<bind>*/from c in NSAlias select 3/*</bind>*/;
-                Diagnostic(ErrorCode.ERR_BadSKunknown, "NSAlias").WithArguments("NSAlias", "namespace").WithLocation(15, 45)
+                Diagnostic(ErrorCode.ERR_BadSKunknown, "NSAlias").WithArguments("ParentNamespace.ConsoleApp", "namespace").WithLocation(15, 45)
```

Comment:
Note that we're now following the alias to the underlying namespace. This matches our behavior in the rest of the compiler.